### PR TITLE
Add NFS source

### PR DESCRIPTION
**Pull request by VladimirSlavik**

### CODE REVIEW

File: pyanaconda/modules/payloads/source/factory.py [hunk 1]
```diff
@@ -46,4 +46,8 @@ def create_source(source_type: SourceType):
                 RepoFilesSourceModule
             return RepoFilesSourceModule()
 
+        elif source_type == SourceType.NFS:
```
Comment (jkonecny12) [inline pyanaconda/modules/payloads/source/factory.py]: Do not add space.

File: pyanaconda/modules/payloads/source/nfs/initialization.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        log.debug("Trying to mount NFS: %s", self._url)
+
+        options, host, path = parse_nfs_url(self._url)
+        if not options:
+            options = "nolock"
+        elif "nolock" not in options:
+            options += ",nolock"
+
+        mount("{}:{}".format(host, path), self._target_mount, fstype="nfs",
+              options=options)
```
Comment (jkonecny12) [inline pyanaconda/modules/payloads/source/nfs/initialization.py]: Please split these argument on separate lines.

File: pyanaconda/modules/payloads/source/nfs/nfs.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        """This source is ready for the installation to start."""
+        ready = os.path.ismount(INSTALL_TREE)
+        log.debug("Source is set to %s ready state.", ready)
+        return ready
+
+    @property
+    def url(self):
+        """URL for mounting.
+
+        Combines server address, path, and options
```
Comment (jkonecny12) [inline pyanaconda/modules/payloads/source/nfs/nfs.py]:
```suggestion
        Combines server address, path, and options.
```

File: tests/nosetests/pyanaconda_tests/module_source_nfs_test.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    def url_empty_properties_test(self):
+        """Test NFS source URL property when not set."""
+        self.assertEqual(self.source_interface.URL, "")
+
+    def url_properties_test(self):
+        """Test NFS source URL property is correctly set."""
+        self.source_interface.SetURL(nfs_url)
+        self.assertEqual(self.source_interface.URL, nfs_url)
+        self.callback.assert_called_once_with(
+            PAYLOAD_SOURCE_NFS.interface_name, {"URL": nfs_url}, [])
```
Comment (jkonecny12) [inline tests/nosetests/pyanaconda_tests/module_source_nfs_test.py]: Please use the `check_dbus_property()` method. See other tests for inspiration how to use it.

File: tests/nosetests/pyanaconda_tests/module_source_nfs_test.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        tasks = self.source_module.set_up_with_tasks()
+
+        # Check the number of the tasks
+        task_number = len(task_classes)
+        self.assertEqual(task_number, len(tasks))
+
+        for i in range(task_number):
+            self.assertIsInstance(tasks[i], task_classes[i])
+
+    def tear_down_with_tasks_test(self):
```
Comment (jkonecny12) [inline tests/nosetests/pyanaconda_tests/module_source_nfs_test.py]: Please use the `check_task_creation_list()` method here.

File: pyanaconda/modules/payloads/source/nfs/initialization.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    def __init__(self, target_mount, url):
+        super().__init__()
+        self._target_mount = target_mount
+        self._url = url
+
+    @property
+    def name(self):
+        return "Set up NFS installation source"
+
+    def run(self):
```
Comment (jkonecny12) [inline pyanaconda/modules/payloads/source/nfs/initialization.py]: You should test if the nfs is already mounted here to avoid re-mounting.

Comment (VladimirSlavik) [reply pyanaconda/modules/payloads/source/nfs/initialization.py]: Done.

File: tests/nosetests/pyanaconda_tests/module_source_nfs_test.py (hunk 5 above)
Comment (jkonecny12) [reply tests/nosetests/pyanaconda_tests/module_source_nfs_test.py]: Please ignore this comment. It's usable only for interface calls.

File: tests/nosetests/pyanaconda_tests/module_source_nfs_test.py (hunk 4 above)
Comment (jkonecny12) [reply tests/nosetests/pyanaconda_tests/module_source_nfs_test.py]: However, this one seems valid to me.

Comment (VladimirSlavik) [reply tests/nosetests/pyanaconda_tests/module_source_nfs_test.py]: Done!

**Pull request merged**